terraform/spec/dependabot/terraform/registry_client_spec.rb

```diff
@@ -37,11 +37,18 @@
 
   it "fetches provider versions", :vcr do
     response = client.all_provider_versions(identifier: "hashicorp/aws")
-    expect(response.max).to eq(Gem::Version.new("3.40.0"))
+    expect(response.max).to eq(Gem::Version.new("3.42.0"))
   end
 
   it "fetches provider versions from a custom registry" do
     hostname = "registry.example.org"
+    stub_request(:get, "https://#{hostname}/.well-known/terraform.json").and_return(
+      status: 200,
+      body: {
+        "modules.v1": "/v1/modules/",
+        "providers.v1": "/v1/providers/"
```

Comment:
```suggestion
        "providers.v1": "https://#{hostname}/v1/providers/"
```

Idea: exercise the relative and absolute paths.

Reply:
I added a separate set of tests so that we could have coverage for absolute and relatives urls. As well as other schemes that we do not support.